src/__Plugin.ino

```diff
@@ -1083,10 +1083,47 @@ byte PluginCall(byte Function, struct EventStruct *event, String& str)
     // Call to all plugins. Return at first match
     case PLUGIN_WRITE:
     case PLUGIN_REQUEST:
-      for (x = 0; x < PLUGIN_MAX; x++)
-        if (Plugin_id[x] != 0)
-          if (Plugin_ptr[x](Function, event, str))
-            return true;
+      {
+        for (byte y = 0; y < TASKS_MAX; y++)
+        {
+          if (Settings.TaskDeviceEnabled[y] && Settings.TaskDeviceNumber[y] != 0)
+          {
+            if (Settings.TaskDeviceDataFeed[y] == 0) // these calls only to tasks with local feed
```

Comment:
Almost the same code is present for the cases:
    case PLUGIN_ONCE_A_SECOND:
    case PLUGIN_TEN_PER_SECOND:
    case PLUGIN_FIFTY_PER_SECOND:
    case PLUGIN_INIT_ALL:
    case PLUGIN_CLOCK_IN:
    case PLUGIN_EVENT_OUT:

I cannot find where TaskDeviceDataFeed is set to something other than 0, so this is called for all enabled plugins.
I guess that other call should use and set the extra added TaskIndex & par4 and 5 also?

Reply:
These changes were made to allow the use of multiple tasks of the same type.

>I cannot find where TaskDeviceDataFeed is set to something other than 0, so this is called for all enabled plugins.

I did not know where TaskDeviceDataFeed was used I kept the same code you saw.

>I guess that other call should use and set the extra added TaskIndex & par4 and 5 also?

TaskIndex, I need to get the task configuration when there are multiple tasks of the same type.
par4 and 5 for PR PCA9685.

Reply:
I can remove 
```c
if (Settings.TaskDeviceEnabled[y] && Settings.TaskDeviceNumber[y] != 0)
```

Reply:
I would always check for enabled tasks, so leave that line in.
The point I was trying to make, was that when you add some elaborate code to add the newly introduced parameters, you may also want to have those new parameters on other locations where similar code is used.
You may not need it now, but if there is a "par5" or "TaskIndex", people will eventually use it.
And if it is not copied the same way, you will get bugs in the future.
So perhaps best to make a proper copy constructor so you don't have to copy every member by hand and only adjust what is different.

I am a bit reluctant to add (and change) members to these events, which are being used almost everywhere in the code, since I cannot foresee what will happen.
So I think it is a good idea to have it tested via a few builds first.  I guess I will merge it first to a separate branch to let it test for a few days.